app/domain/authentication/authn_jwt/validate_and_decode/validate_and_decode_token.rb

```diff
@@ -7,18 +7,7 @@ module ValidateAndDecode
       # for the 2nd validation
       ValidateAndDecodeToken ||= CommandClass.new(
         dependencies: {
-          fetch_signing_key: ::Util::ConcurrencyLimitedCache.new(
-            ::Util::RateLimitedCache.new(
-              ::Authentication::AuthnJwt::SigningKey::FetchCachedSigningKey.new(
-                signing_key_interface: ::Authentication::AuthnJwt::SigningKey::FetchSigningKeyInterface
-              ),
-              refreshes_per_interval: CACHE_REFRESHES_PER_INTERVAL,
-              rate_limit_interval: CACHE_RATE_LIMIT_INTERVAL,
-              logger: Rails.logger
-            ),
-            max_concurrent_requests: CACHE_MAX_CONCURRENT_REQUESTS,
-            logger: Rails.logger
-          ),
+          fetch_signing_key: ::Authentication::AuthnJwt::SigningKey::FetchSigningKeyInterface,
```

Comment:
how does this work? when is the `fetch_signing_key` instance created?

Reply:
Moving it to input

Reply:
Its created on vendor configuration when validate_and_decode is called as a pre step before validate and decode

Reply:
We need this dependency for testing. 
This is created on vendor configuration as pre step when validate_and_decode is being called

Reply:
this is actually the exact opposite of what we want. we use dependencies as default values for app code and inject mock objects in UTs. please move this to be an input and pass it as an input in app code and in UTs